src/cli/mod.rs

```diff
@@ -0,0 +1,22 @@
+use crate::error::Error;
+use async_trait::async_trait;
+use clap::Args;
+
+/// The trait that represents the abstraction of a command line handler. To implement a new command
+/// line operation, implement this trait and register it, see `register::register_cli_command` for
+/// how to register.
+///
+/// Note that this trait does not support a stateful implementation as we assume CLI commands are all
+/// constructed from scratch. Initialize the states in the `handle` method.
+#[async_trait]
+pub trait CommandLineHandler {
+    /// The request to process.
```

Comment:
```suggestion
    /// Abstraction for command line operations.
```
Let's try to avoid the name "request", let's use command line, arguments or method.